host-controller/src/main/java/org/jboss/as/host/controller/model/jvm/JvmOptionsBuilderFactory.java

```diff
@@ -167,6 +168,19 @@ boolean checkOption(boolean condition, String jvm, String option, String schemaE
     }
 
     private static class SunJvmOptionsBuilder extends JvmOptionsBuilder {
+        public static final int JVM_MAJOR_VERSION;
+
+        static {
+            int vmVersion;
+            try {
+                String vmVersionStr = WildFlySecurityManager.getPropertyPrivileged("java.specification.version", null);
+                vmVersion = Integer.valueOf(vmVersionStr.substring(2));
```

Comment:
You are making an assumption here.  You should validate the assumption (using a regex is probably simplest) before extracting the minor version number.  If the assumption cannot be validated, then you need to have some sane recourse (even if it's just throwing an exception), because otherwise this will become one of those impossible-to-figure-out bugs when Oracle suddenly decides that it's time for Java 2.0.